extensions/jaxb/deployment/src/main/java/io/quarkus/jaxb/deployment/JaxbProcessor.java

```diff
@@ -170,18 +172,73 @@ void processAnnotationsAndIndexFiles(
         JAXB_ANNOTATIONS.stream()
                 .map(Class::getName)
                 .forEach(className -> {
-                    proxyDefinitions.produce(new NativeImageProxyDefinitionBuildItem(className, Locatable.class.getName()));
+                    proxyDefinitions.produce(new NativeImageProxyDefinitionBuildItem(className,
+                            "com.sun.xml.bind.v2.model.annotation.Locatable"));
                     addReflectiveClass(reflectiveClass, true, false, className);
                 });
 
+        proxyDefinitions
+                .produce(new NativeImageProxyDefinitionBuildItem("com.sun.xml.bind.marshaller.CharacterEscapeHandler"));
+        proxyDefinitions
+                .produce(new NativeImageProxyDefinitionBuildItem("com.sun.xml.txw2.output.CharacterEscapeHandler"));
+
+        proxyDefinitions.produce(new NativeImageProxyDefinitionBuildItem("com.sun.xml.txw2.TypedXmlWriter"));
+
         for (JaxbFileRootBuildItem i : fileRoots) {
-            try (Stream<Path> stream = iterateResources(i.getFileRoot())) {
+            try (Stream<Path> stream = iterateResources(applicationArchivesBuildItem, i.getFileRoot())) {
                 stream.filter(p -> p.getFileName().toString().equals("jaxb.index"))
                         .forEach(p1 -> handleJaxbFile(p1, resource, reflectiveClass));
             }
         }
     }
 
+    @BuildStep
+    void processTypedXmlWriters(
+            CombinedIndexBuildItem combinedIndexBuildItem,
+            BuildProducer<NativeImageProxyDefinitionBuildItem> proxyDefinitions) {
+        IndexView index = combinedIndexBuildItem.getIndex();
+        DotName typedXmlWriterDN = DotName.createSimple("com.sun.xml.txw2.TypedXmlWriter");
+        // getAllKnownDirectImplementors skip interface, so use own recursion
+        produceRecursiveProxies(index, typedXmlWriterDN)
+                .map(DotName::toString)
+                .map(NativeImageProxyDefinitionBuildItem::new)
+                .forEach(proxyDefinitions::produce);
+    }
+
+    @BuildStep
+    void seeAlso(CombinedIndexBuildItem combinedIndexBuildItem,
+            BuildProducer<ReflectiveClassBuildItem> reflectiveItems) {
+        IndexView index = combinedIndexBuildItem.getIndex();
+        for (AnnotationInstance xmlSeeAlsoAnn : index.getAnnotations(XML_SEE_ALSO)) {
+            AnnotationValue value = xmlSeeAlsoAnn.value();
+            Type[] types = value.asClassArray();
+            for (Type t : types) {
+                reflectiveItems.produce(new ReflectiveClassBuildItem(false, false, t.name().toString()));
+            }
+        }
+    }
+
+    public static boolean ModifierIsInterface(ClassInfo classinfo) {
+        return Modifier.isInterface(classinfo.flags());
+    }
+
+    Stream<DotName> produceRecursiveProxies(IndexView index,
```

Comment:
This method isn't producing anything, just collecting DotNames

Reply:
yes that s the purpose because it is a tree of inheritance with interface that I need to produce NativeImageProxyDefinitionBuildItem for each interface.
We have 2 solutions:
- list all class and hardcode the class from jaxbri. it can be found there : https://github.com/quarkiverse/quarkus-cxf/commit/8ee8b23ee1625e31036560bceb6c2cb83feaa755#diff-ab6082b1f5a32014ec91e75ab14715d38b1d3e2bd2cab287562542c1af6a37b6L1577-L1623 (about 50 classes)
- do this inheritance calculation on build time with recurssive.

Reply:
Both solutions are OK, but I'd prefer to have the classes hardcoded from the jaxb-ri in this case instead for these reasons:  

- That would make it more deterministic (and faster than calculating the inheritance on build time  that we know it will never change).
- You can simplify the Processor by not indexing the JAXB-RI dependency

Reply:
to avoid computation, I have just provide list of class.